Deleting projects returns HTTP 500 and fails to delete quotas

**Expected behavior and actual behavior:**
Expected: Deleting a project should complete successfully
Actual: Delete returns HTTP 500

**Steps to reproduce the problem:**
Create a project
Delete the project

**Versions:**
Please specify the versions of following systems.

- harbor version: [1.10.3]
- harbor helm: [1.3.3]
- GKE

**Additional context:**

Logs indicate:
```
2020-06-19T16:16:00Z [WARNING] [/common/quota/manager.go:278]: Load quota reference object (project, 83) failed: project not found, project_id: 83
2020-06-19T16:16:00Z [ERROR] [/common/api/base.go:233]: failed to get quota manager: project not found, project_id: 83
2020-06-19T16:16:00Z [ERROR] [/common/api/base.go:73]: DELETE /api/projects/83 failed with error: {"code":500,"message":"Internal Server Error"}
2020-06-19 12:16:00.030 EDT remote_addr - "DELETE /api/projects/83 HTTP/1.1" 500 47 "-" "-" 0.102 0.102 .
```

It looks like the project model is deleted, and then the associated quota manager is loaded:
https://github.com/goharbor/harbor/blob/v1.10.3/src/core/api/project.go#L295-L300
but the quota's loader driver tries to load the project model again:
https://github.com/goharbor/harbor/blob/v1.10.3/src/common/quota/driver/project/driver.go#L53

--- a/src/common/quota/driver/project/driver.go
+++ b/src/common/quota/driver/project/driver.go
@@ -50,7 +50,7 @@ func getProjectsBatchFn(ctx context.Context, keys dataloader.Keys) []*dataloader
 		projectIDs = append(projectIDs, id)
 	}
 
-	projects, err := dao.GetProjects(&models.ProjectQueryParam{})
+	projects, err := dao.GetProjects(&models.ProjectQueryParam{ProjectIDs: projectIDs})
 	if err != nil {
 		return handleError(err)
 	}
@@ -154,7 +154,7 @@ func (d *driver) Validate(hardLimits types.ResourceList) error {
 func newDriver() dr.Driver {
 	cfg := config.NewDBCfgManager()
 
-	loader := dataloader.NewBatchedLoader(getProjectsBatchFn)
+	loader := dataloader.NewBatchedLoader(getProjectsBatchFn, dataloader.WithClearCacheOnBatch())
 
 	return &driver{cfg: cfg, loader: loader}
 }

--- a/src/core/api/project.go
+++ b/src/core/api/project.go
@@ -292,11 +292,6 @@ func (p *ProjectAPI) Delete() {
 		return
 	}
 
-	if err = p.ProjectMgr.Delete(p.project.ProjectID); err != nil {
-		p.ParseAndHandleError(fmt.Sprintf("failed to delete project %d", p.project.ProjectID), err)
-		return
-	}
-
 	quotaMgr, err := quota.NewManager("project", strconv.FormatInt(p.project.ProjectID, 10))
 	if err != nil {
 		p.SendInternalServerError(fmt.Errorf("failed to get quota manager: %v", err))
@@ -304,6 +299,11 @@ func (p *ProjectAPI) Delete() {
 	}
 	if err := quotaMgr.DeleteQuota(); err != nil {
 		p.SendInternalServerError(fmt.Errorf("failed to delete quota for project: %v", err))
+		return
+	}
+
+	if err = p.ProjectMgr.Delete(p.project.ProjectID); err != nil {
+		p.ParseAndHandleError(fmt.Sprintf("failed to delete project %d", p.project.ProjectID), err)
 		return
 	}
 

